src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -94,45 +88,41 @@ private function registerHandlers(ContainerBuilder $container, array $busIds)
                     $handles = $this->guessHandledClasses($r, $serviceId);
                 }
 
-                $priority = $tag['priority'] ?? 0;
+                $message = null;
                 $handlerBuses = (array) ($tag['bus'] ?? $busIds);
 
-                foreach ($handles as $messageClass => $method) {
+                foreach ($handles as $message => $method) {
                     $buses = $handlerBuses;
-                    if (\is_int($messageClass)) {
-                        $messageClass = $method;
+                    if (\is_int($message)) {
+                        $message = $method;
                         $method = '__invoke';
                     }
 
-                    if (\is_array($messageClass)) {
-                        $messagePriority = $messageClass[1];
-                        $messageClass = $messageClass[0];
+                    if (\is_array($message)) {
+                        list($message, $priority) = $message;
```

Comment:
this case is not documented and seems to be a relict of the old subscriber return values. considering all the bc breaks, this could be removed as well I guess.

Reply:
I'd suggest doing so in another PR - here it's unrelated (and I feel like this compiler pass needs to be improved anyway)